packages/flutter/lib/src/material/reorderable_list.dart

```diff
@@ -581,10 +590,25 @@ class _ReorderableListContentState extends State<_ReorderableListContent> with T
         reverse: widget.reverse,
         child: _buildContainerForScrollDirection(
           children: <Widget>[
-            if (widget.reverse) _wrap(finalDropArea, widget.children.length, constraints),
+            // 0 must be the first index at the start of the reversed list,
+            // so that items which get dragged to the start are ordered to start
+            if (widget.reverse) _wrap(finalDropArea, 0, constraints),
             if (widget.header != null) widget.header,
             for (int i = 0; i < widget.children.length; i += 1) _wrap(widget.children[i], i, constraints),
-            if (!widget.reverse) _wrap(finalDropArea, widget.children.length, constraints),
+            if (widget.footer != null)
+              // We need to wrap the widget.footer so we can reorder an item between
+              // the last item of the list and the footer itself
+              _wrap(SizedBox(
+                  key: Key('footer'),
```

Comment:
Why is adding the key here necessary?

Reply:
I added footer locally and I encountered, that the footer also has to have key. Don't know why but it seems to ReorderableListView checks for it.